### PR TITLE
Refactor the API views

### PR DESCRIPTION
**Pull request by cc7768**
This PR refactors the API views. It does a few main changes:

* Separates `api.economics` and `api.economic_snapshots` and highlights that the latter is focused on data that is slow moving and is more about the characteristics of an area (such as the BEA GDP tables) while the former is focused on data that we can get at a daily/weekly/monthly/quarterly basis.
* Creates a separate view for the NYT Covid data. Eventually we will tie these together and determine a priority ordering among data sources but it seems worth it to expose the NYT data (and eventually the JHU and USA Facts datasets) separately.
* Renames the `fips` column to `location` in the API views... We will probably want to make this change at the data/meta level at some point, but, in preparation for that, it made sense to make it here before publicizing.
* No longer using `meta_date` columns for `demographics` and `economic_snapshots` -- Instead we're using the most recent data. In the `demographics` view, I'm assuming that the `name` from `acs_variables_selected` is unique -- @sglyon would you add this uniqueness constraint to the `meta.acs_variables_selected` table and add its definition to this PR for reference.

### CODE REVIEW

**Comment by sglyon** [comment]
@cc7768 I don't see the breakout into mulitple covid tables here. I believe at one point you were planning to have a `covid_testing` `covid_healthcare`, etc views. Is that right? If so, did you decide against it?

>  @sglyon would you add this uniqueness constraint to the meta.acs_variables_selected table and add its definition to this PR for reference

I can add sql here, but I'm not sure what exactly you are wanting. It may be easiest if you hop on pgadmin, highlight the table in question and then press the SQL tab and finding the sql you want

pgAdmin 4

**Comment by sglyon** [comment]
Ahh, I didn't understand at the time, but I think I do now. 

was that a request that I create that constraint and put the corresponding SQL in this PR?

**Review by cc7768 — COMMENTED** [review]
Ok. I think I've addressed most of what should be taken care of (and more...) in this PR. Let's see if we can't get this and the other related PRs merged.

File: schemas/api_demographics.sql [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
-    dd.value
-   FROM data.acs_data dd
-     LEFT JOIN meta.acs_variables mv ON dd.id = mv.id
-     LEFT JOIN meta.acs_variables_selected mvs ON mv.year = mvs.year AND mv.product = mvs.product AND mv.census_id::text = mvs.census_id::text;
-
-
-COMMENT ON VIEW api.demographics IS E'This table contains information on the demographics of a particular region and is based on the American Community Survey.
-
-Currently, the following variables are collected in the database
+/* TODO: Need to add a unique constraint for name in the acs_variables_selected table */
```
Comment (cc7768) [inline schemas/api_demographics.sql]:
```suggestion
```

File: schemas/api_economics.sql [hunk 2]
```diff
@@ -1,26 +1,30 @@
...
-    weeklyeconomicindex.wei AS value
-   FROM data.weeklyeconomicindex;
+CREATE OR REPLACE VIEW api.economics AS
+  WITH last_vintage AS (
+    SELECT MAX(vintage), dt, fips, variable_name
+    FROM dol_ui
+    GROUP BY dt, fips, variable_name
+  )
+  SELECT dt, fips as location, variable_name as variable, value
+  FROM dol_ui ui
```
Comment (sglyon) [inline schemas/api_economics.sql:8]: needs to be `data.dol_ui`

File: schemas/api_covid.sql [hunk 3]
```diff
@@ -1,19 +1,85 @@
...
+  FROM last_vintage lv
+  LEFT JOIN data.jhu_covid nyt using (fips, variable_id, vintage)
+  LEFT JOIN meta.covid_variables cv ON cv.id = uc.variable_id;
+*/
+
+CREATE OR REPLACE VIEW api.nytimes_covid AS
+  WITH last_vintage as (
+    SELECT fips, variable_id, MAX(vintage) AS vintage
+    FROM data.nyt_covid
+    GROUP BY (fips, variable_id)
```
Comment (cc7768) [inline schemas/api_covid.sql]:
```suggestion
    GROUP BY (dt, fips, variable_id)
```

Comment (sglyon) [inline schemas/api_covid.sql]: lets add `dt` as part of the groupby. will also need to join with it below

Comment (cc7768) [reply schemas/api_covid.sql]: Add `dt` to groupby so that we don't end up with conflicts

File: schemas/api_covid.sql [hunk 4]
```diff
@@ -1,19 +1,85 @@
...
+  )
+  SELECT jhu.dt, jhu.fips, cv.name as variable, jhu.value
+  FROM last_vintage lv
+  LEFT JOIN data.jhu_covid nyt using (fips, variable_id, vintage)
+  LEFT JOIN meta.covid_variables cv ON cv.id = uc.variable_id;
+*/
+
+CREATE OR REPLACE VIEW api.nytimes_covid AS
+  WITH last_vintage as (
+    SELECT fips, variable_id, MAX(vintage) AS vintage
```
Comment (cc7768) [inline schemas/api_covid.sql]:
```suggestion
    SELECT dt, fips, variable_id, MAX(vintage) AS vintage
```

File: schemas/api_covid.sql [hunk 5]
```diff
@@ -1,19 +1,85 @@
...
+
+CREATE OR REPLACE VIEW api.nytimes_covid AS
+  WITH last_vintage as (
+    SELECT dt, fips, variable_id, MAX(vintage) AS vintage
+    FROM data.nyt_covid
+    GROUP BY (dt, fips, variable_id)
+  )
+  SELECT nyt.dt, nyt.fips, cv.name as variable, nyt.value
+  FROM last_vintage lv
+  LEFT JOIN data.nyt_covid nyt using (fips, variable_id, vintage)
```
Comment (cc7768) [inline schemas/api_covid.sql]:
```suggestion
  LEFT JOIN data.nyt_covid nyt using (dt, fips, variable_id, vintage)
```

**Review by sglyon — APPROVED** [review]

**Pull request merged**